src/Model/MyParcelRequest.php

```diff
@@ -98,16 +98,22 @@ public function getError()
      * @param string $body
      * @param string $requestHeader
      *
+     * @param bool   $sendMail
+     *
      * @return $this
      */
-    public function setRequestParameters($apiKey, $body, $requestHeader): MyParcelRequest
+    public function setRequestParameters(string $apiKey, string $body, string $requestHeader, bool $sendMail = true): MyParcelRequest
     {
         $this->api_key = $apiKey;
         $this->body    = $body;
 
         $header[] = $requestHeader;
         $header[] = 'Authorization: basic ' . base64_encode($this->api_key);
 
+        if (! $sendMail) {
+            $header[] = 'send_return_mail: ' . $sendMail;
+        }
+
         $this->header = $header;
```

Comment:
Waarom doe je hier niet zoiets:

`$requestHeader` -> `$headers`
`$this->header` -> `$this->headers`

```php
$this->headers = [
    'Authorization' => 'basic ' . base64_encode($this->api_key),
] + $headers;

if (! $sendMail) {
    $this->headers['send_return_mail'] = $sendMail;
}
```
Dan moet je wel waar er headers meegestuurd worden even het format veranderen in `[a => b]` en waar de headers gebruikt worden het zelf weer even omgooien naar een array van strings.

Ik zie ook de default headers en het sendmail header stukjes graag allebei in een eigen functie, maar als je dat niet ziet zitten vind ik het niet erg :P

Reply:
Het is nu anders, ik stuur het nu als een url parameter mee